### PR TITLE
Initial skeleton for the safehttp library that integrates the net/http package

### PR DESCRIPTION
**Pull request by mihalimara22**
Created a dummy structure for safehttp.IncomingRequest and safehttp.ResponseWriter that integrates http.ResponseWriter and http.Request

### CODE REVIEW

File: safehttp/prototype.go [hunk 1]
```diff
@@ -12,7 +16,10 @@ func NewMachinery(h HandleFunc, d Dispatcher) *Machinery {
 }
 
 // HandleRequest TODO
-func (m *Machinery) HandleRequest(r string) {
-	rw := &ResponseWriter{d: m.d}
-	m.h(rw, nil)
+func (m *Machinery) HandleRequest(w http.ResponseWriter, req *http.Request) {
+	rw := ResponseWriter{d: m.d, rw: w}
+
```
Comment (kele) [inline safehttp/prototype.go:21]: nit: get rid of the newlines.

**Comment by mattiasgrenfeldt** [comment]
Rebased on top of master

File: safehttp/response_writer.go [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
 }
 
 // Result TODO
 type Result struct{}
 
 // Write TODO
 func (w *ResponseWriter) Write(resp Response) Result {
-	err := w.d.Write(resp)
+	err := w.d.Write(w.rw, resp)
 	if err != nil {
```
Comment (empijei) [inline safehttp/response_writer.go:19]:
```suggestion
	if err := w.d.Write(w.rw, resp); err != nil {
```

File: safehttp/response_writer.go [hunk 3]
```diff
@@ -19,7 +24,7 @@ func (w *ResponseWriter) Write(resp Response) Result {
 
 // WriteTemplate TODO
 func (w *ResponseWriter) WriteTemplate(t Template, data interface{}) Result {
-	err := w.d.ExecuteTemplate(t, data)
+	err := w.d.ExecuteTemplate(w.rw, t, data)
 	if err != nil {
```
Comment (empijei) [inline safehttp/response_writer.go:28]:
```suggestion
	if err := w.d.ExecuteTemplate(w.rw, t, data); err != nil {
```

File: tests/integration/safehtml/safehtml_test.go [hunk 4]
```diff
@@ -1,70 +1,79 @@
...
 	"github.com/google/safehtml/template"
 )
 
-type dispatcher struct {
-	wr io.Writer
-}
+type dispatcher struct{}
 
-func (d *dispatcher) Write(resp safehttp.Response) error {
+func (d *dispatcher) Write(rw http.ResponseWriter, resp safehttp.Response) error {
```
Comment (empijei) [inline tests/integration/safehtml/safehtml_test.go:16]:
```suggestion
func (*dispatcher) Write(rw http.ResponseWriter, resp safehttp.Response) error {
```

File: tests/integration/safehtml/safehtml_test.go [hunk 5]
```diff
@@ -1,70 +1,79 @@
...
-		_, err := d.wr.Write([]byte(x.String()))
+		_, err := rw.Write([]byte(x.String()))
 		return err
 	default:
 		panic("not a safe response type")
 	}
 }
 
-func (d *dispatcher) ExecuteTemplate(t safehttp.Template, data interface{}) error {
+func (d *dispatcher) ExecuteTemplate(rw http.ResponseWriter, t safehttp.Template, data interface{}) error {
```
Comment (empijei) [inline tests/integration/safehtml/safehtml_test.go:26]:
```suggestion
func (*dispatcher) ExecuteTemplate(rw http.ResponseWriter, t safehttp.Template, data interface{}) error {
```

File: tests/integration/safehtml/safehtml_test.go [hunk 6]
```diff
@@ -1,70 +1,79 @@
...
 		panic("not a safe response type")
 	}
 }
 
 func TestHandleRequestWrite(t *testing.T) {
-	b := &strings.Builder{}
-	d := &dispatcher{wr: b}
+	m := safehttp.NewMachinery(func(rw safehttp.ResponseWriter, _ *safehttp.IncomingRequest) safehttp.Result {
+		return rw.Write(safehtml.HTMLEscaped("<h1>Escaped, so not really a heading</h1>"))
+	}, &dispatcher{})
```
Comment (empijei) [inline tests/integration/safehtml/safehtml_test.go:38]: Don't @ me, but I am starting to think the dispatcher should somehow be registered in the machinery and not passed every time a user of the package needs to write a response.

Comment (kele) [reply tests/integration/safehtml/safehtml_test.go:38]: It is registered in the machinery. `NewMachinery` is a constructor here.

Comment (empijei) [reply tests/integration/safehtml/safehtml_test.go:38]: I misread the parentheses, disregard my comment pls.

**Pull request merged**